experiments/quickcell/quickcell_rectangular_gate_library.cpp

```diff
@@ -0,0 +1,175 @@
+         {std::vector<tt>{create_le_tt()}, "le_2i_top_left_1o_right"},
+         {std::vector<tt>{create_ge_tt()}, "ge_2i_top_left_1o_right"},
+         {std::vector<tt>{create_crossing_wire_tt()}, "cx_2i_top_left_2o_down_right"},
+         {std::vector<tt>{create_half_adder_tt()}, "ha_2i_top_left_2o_down_right"},
+         {std::vector<tt>{create_double_wire_tt()}, "hourglass_2i_top_left_2o_down_right"}}};
+
+    static const std::string folder = fmt::format("{}/gate_skeletons/rectangular_skeletons/", EXPERIMENTS_PATH);
+
+    const auto rectangular_2i_top_left_2o_down_right = read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(
+        fmt::format("{}/{}", folder, "rectangular_2i_top_left_2o_down_right.sqd"));
+
+    const auto rectangular_2i_top_left_1o_right = read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(
+        fmt::format("{}/{}", folder, "rectangular_2i_top_left_1o_right.sqd"));
+
+    const auto rectangular_1i_top_1o_right =
+        read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(fmt::format("{}/{}", folder, "rectangular_1i_top_1o_right.sqd"));
+
+    const auto rectangular_1i_top_1o_down =
+        read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(fmt::format("{}/{}", folder, "rectangular_1i_top_1o_down.sqd"));
+
+    const auto rectangular_1i_left_1o_right =
+        read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(fmt::format("{}/{}", folder, "rectangular_1i_left_1o_right.sqd"));
+
+    const auto rectangular_1i_top_2o_right_down = read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(
+        fmt::format("{}/{}", folder, "rectangular_1i_top_2o_right_down.sqd"));
+
+    const auto rectangular_1i_top_2o_left_right = read_sqd_layout<sidb_100_cell_clk_lyt_siqad>(
+        fmt::format("{}/{}", folder, "rectangular_1i_top_2o_left_right.sqd"));
+
+    const auto num_canvas_sidbs = 3;
+
+    design_sidb_gates_params<fiction::cell<sidb_100_cell_clk_lyt_siqad>> params{
+        is_operational_params{sidb_simulation_parameters{2, -0.32}, sidb_simulation_engine::QUICKEXACT,
+                              bdl_input_iterator_params{{3}}, operational_condition::REJECT_KINKS},
+        design_sidb_gates_params<fiction::cell<sidb_100_cell_clk_lyt_siqad>>::design_sidb_gates_mode::QUICKCELL,
+        {{18, 9, 0}, {26, 13, 0}},
+        num_canvas_sidbs};
+
+    for (const auto& [truth_table, gate_name] : truth_tables_and_names)
+    {
+        std::vector<sidb_100_cell_clk_lyt_siqad> quickcell_design{};
+        design_sidb_gates_stats                  stats_quickcell{};
+
+        if (gate_name == "cx_2i_top_left_2o_down_right" || gate_name == "ha_2i_top_left_2o_down_right" ||
+            gate_name == "hourglass_2i_top_left_2o_down_right")
+        {
+            params.number_of_sidbs = 4;
+            params.canvas          = {{17, 8, 0}, {27, 14, 0}};
+            quickcell_design =
+                design_sidb_gates(rectangular_2i_top_left_2o_down_right, truth_table, params, &stats_quickcell);
+            params.number_of_sidbs = num_canvas_sidbs;
```

Comment:
Instead of resetting the number of SiDBs here, I would explicitly set them below. Do avoid code duplication, you could create one big `else` in which you set the value once, and have several `if ... else if` inside.